Add symfony events for login action

sharidas opened the pull request:
Add symfony events for login action.
The symfony events are added for:
1 - login with token
2 - login with apache

Signed-off-by: Sujith H <sharidasan@owncloud.com>

<!--
Thanks for submitting a change to ownCloud!

This is the bug tracker for the Server component. Find other components at https://github.com/owncloud/core/blob/master/.github/CONTRIBUTING.md#guidelines

For fixing potential security issues please see https://owncloud.org/security/

To make it possible for us to get your change reviewed and merged please fill out below information carefully.

Please note that any kind of change first has to be submitted to the master branch which holds the next major version of ownCloud.

We will carefully discuss if your change can or has to be backported to stable branches.
-->

## Description
<!--- Describe your changes in detail -->
Adding symfony events for the missing login methods.

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
https://github.com/owncloud/core/issues/31956

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
Adding symfony events for the missing login methods.

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
[x] Tested by adding unit tests.

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

sharidas committed:
Add symfony events for login action

Add symfony events for login action.
The symfony events are added for:
1 - login with token
2 - login with apache

Signed-off-by: Sujith H <sharidasan@owncloud.com>

sharidas commented:
The reason for using `login` instead of `uid` is because of https://github.com/owncloud/core/blob/master/lib/private/User/Session.php#L505.

PVince81 reviewed (commented):
This looks good.

I wonder if we should use the opportunity to add another argument "$loginType" that tells whether login was done with password, with token or apache in case it would be useful for apps.

sharidas commented:
> I wonder if we should use the opportunity to add another argument "$loginType" that tells whether login was done with password, with token or apache in case it would be useful for apps.

For the event listeners it would be an added advantage. I will add the change to this PR. Thanks.

sharidas commented:
Added :
`'loginType' => 'password'`, `'loginType' => 'token'` and `'loginType' => 'apache'`
Also updated the tests so that mocked EventDispatcher is used, to improve readability of the test code.

PVince81 commented on tests/lib/User/SessionTest.php:
```diff
@@ -1540,6 +1586,15 @@ function (GenericEvent $event) use (&$calledFailedLoginEvent) {
 			->willReturn('foo');
 
+		$failedEvent = new GenericEvent(null, ['user' => 'foo']);
+		//$beforeloginEvent = new GenericEvent(null, ['user' => $iUser, 'uid' => 'foo', 'password' => 'foo']);
```
remove?

sharidas replied:
Yes need to remove this. Removed in the updated version.

PVince81 commented on tests/lib/User/SessionTest.php:
```diff
@@ -1540,6 +1586,15 @@ function (GenericEvent $event) use (&$calledFailedLoginEvent) {
+		$eventDispatcher->expects($this->once())
+			->method('dispatch')
+			->withConsecutive(
+				//[$this->equalTo('user.beforelogin'), $this->equalTo($beforeloginEvent)],
```
remove ? or is this not covered yet ?

sharidas replied:
Yes need to remove this line.

PVince81 reviewed (commented):
Thanks for the addition.

Please remove or integrate the commented out test code

PVince81 commented:
@butonic @tomneedham FYI https://github.com/owncloud/core/pull/31963#issuecomment-402444259

PVince81 reviewed (approved):
:+1:

sharidas commented:
Backport PR: https://github.com/owncloud/core/pull/31985
